Apps/GatewayApi/src/Models/AssociateDelegationRequest.cs

```diff
@@ -0,0 +1,33 @@
+//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.GatewayApi.Models
+{
+    /// <summary>
+    /// The AssociateDelegationRequest model.
+    /// </summary>
+    public record AssociateDelegationRequest
+    {
+        /// <summary>
+        /// Gets or sets the profile hdid.
+        /// </summary>
+        public string ProfileHdid { get; set; } = string.Empty;
```

Comment:
I'm looking to consume this now from the front end but I've realised you have two hdid's here:

1. URL path parameter 
2. profileHdid

can we not just submit the current user's hdid (delegate) and the encryptedDelegationId?

Reply:
Yes, this is being changed to remove the model and pass the encryptedDelegationId as a query parameter.  There are some additional changes being made in the service and validators to reference the right HDIDs as well.

Reply:
Code changed as per discussion with @MikeLyttle